### PR TITLE
Add tests for room name

### PR DESCRIPTION
**Pull request by bmarty**
Also:
- cleanup the SDK
- cleanup the opened sessions during the tests

Note that the room name tests succeed only if executed 1 by 1...

### CODE REVIEW

File: matrix-sdk/src/androidTest/java/org/matrix/androidsdk/common/CommonTestHelper.java [hunk 1]
```diff
@@ -355,4 +357,46 @@ public void onSuccess(Credentials credentials) {
     public void await(CountDownLatch latch) throws InterruptedException {
         Assert.assertTrue(latch.await(TestConstants.AWAIT_TIME_OUT_MILLIS, TimeUnit.MILLISECONDS));
     }
+
+    /**
+     * Clear all non null sessions in lazy loading scenario data
+     *
+     * @param data
+     */
+    public void clearAllSessions(LazyLoadingScenarioData data) {
```
Comment (ganfra) [inline matrix-sdk/src/androidTest/java/org/matrix/androidsdk/common/CommonTestHelper.java]: Don't add this to Common as it's belong to lazy loading

File: matrix-sdk/src/androidTest/java/org/matrix/androidsdk/common/CommonTestHelper.java [hunk 2]
```diff
@@ -355,4 +357,46 @@ public void onSuccess(Credentials credentials) {
...
+
+        clearAllSessions(sessionsToClear);
+    }
+
+    /**
+     * Clear all sessions in room name scenario data
+     *
+     * @param data
+     */
+    public void clearAllSessions(RoomNameScenarioData data) {
```
Comment (ganfra) [inline matrix-sdk/src/androidTest/java/org/matrix/androidsdk/common/CommonTestHelper.java]: Same as LazyLoading comment

File: matrix-sdk/src/androidTest/java/org/matrix/androidsdk/common/CommonTestHelper.java (hunk 1 above)
Comment (bmarty) [reply matrix-sdk/src/androidTest/java/org/matrix/androidsdk/common/CommonTestHelper.java]: done

File: matrix-sdk/src/androidTest/java/org/matrix/androidsdk/common/CommonTestHelper.java (hunk 2 above)
Comment (bmarty) [reply matrix-sdk/src/androidTest/java/org/matrix/androidsdk/common/CommonTestHelper.java]: done

**Comment by bmarty** [comment]
Up @ganfra

**Pull request merged**